Zend/Optimizer/zend_optimizer.c

```diff
@@ -831,6 +831,49 @@ zend_class_entry *zend_optimizer_get_class_entry_from_op1(
 	return NULL;
 }
 
+const zend_class_constant *zend_fetch_class_const_info(
+	const zend_script *script, const zend_op_array *op_array, const zend_op *opline, bool *is_prototype) {
+	const zend_class_entry *ce = NULL;
+	bool is_static_reference = false;
+
+	if (!opline || !op_array || !script || opline->op2_type != IS_CONST || Z_TYPE(ZEND_OP2_LITERAL(opline)) != IS_STRING) {
+		return NULL;
+	}
+	if (opline->op1_type == IS_CONST) {
+		zval *op1 = CRT_CONSTANT(opline->op1);
+		if (Z_TYPE_P(op1) == IS_STRING) {
+			ce = zend_optimizer_get_class_entry(script, op_array, Z_STR_P(op1 + 1));
+		}
+	} else if (opline->op1_type == IS_UNUSED && op_array->scope && !(op_array->scope->ce_flags & ZEND_ACC_TRAIT)) {
+		int fetch_type = (int) opline->op1.num & ZEND_FETCH_CLASS_MASK;
```

Comment:
Shouldn't need a cast?

Reply:
Sure, it doesn't need it, so I'm removing it. I added the case only because Clion nagged me with `lang-Tidy: Narrowing conversion from 'uint32_t' (aka 'unsigned int') to signed type 'int' is implementation-definied`. I know it's a false positive, but I wanted to avoid the yellow underline... ^^